crypto/bio/bss_file.c

```diff
@@ -142,13 +142,13 @@ static int file_read(BIO *b, char *out, int outl)
     int ret = 0;
 
     if (b->init && (out != NULL)) {
-        if (b->flags & BIO_FLAGS_UPLINK)
+        if (b->flags & BIO_FLAGS_UPLINK_INTERNAL)
             ret = UP_fread(out, 1, (int)outl, b->ptr);
         else
             ret = fread(out, 1, (int)outl, (FILE *)b->ptr);
         if (ret == 0
-            && (b->flags & BIO_FLAGS_UPLINK) ? UP_ferror((FILE *)b->ptr) :
-                                               ferror((FILE *)b->ptr)) {
+            && (b->flags & BIO_FLAGS_UPLINK_INTERNAL)
```

Comment:
Funny, I think the brackets are misplaced here.
isn't this effectively `if ( (ret == 0 && b->flags & BIO_FLAGS_UPLINK) ? UP_ferror (b->ptr) : ferror(b->ptr) ) {`